delegate: fix SHOW ENUMS column names

otan opened the pull request:
Fixes a bug in #54708. 

Release note (bug fix): Fix SHOW ENUMS column names to have `values`
instead of `string_agg` for column names, and `owner` for the owner
itself.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55139)
<!-- Reviewable:end -->

otan committed:
delegate: fix SHOW ENUMS column names

Release note (bug fix): Fix SHOW ENUMS column names to have `values`
instead of `string_agg` for column names, and `owner` for the owner
itself.

Release note (bug fix): Fix SHOW TYPES to show owner column instead of
the value column.

ajstorm reviewed (approved):
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55139)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm and @otan)

___
*[pkg/sql/delegate/show_types.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55139#-MI_sjM2EVhOYrM9cr3x:-MI_sjM2EVhOYrM9cr3y:b-mdpna6) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d81479bece39e872045ddf2580a91626bc3dc5f/pkg/sql/delegate/show_types.go#L20)):*
> ```Go
> 	return parse(`
> SELECT
>   schema, name, value AS description
> ```

You thinking that we don't need to return the values here?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/sql/delegate/show_types.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55139#-MI_sjM2EVhOYrM9cr3x:-MI_t9lf0jxOnBYkBVa4:bmcvibm) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d81479bece39e872045ddf2580a91626bc3dc5f/pkg/sql/delegate/show_types.go#L20)):*
<details><summary><i>Previously, ajstorm wrote…</i></summary><blockquote>

You thinking that we don't need to return the values here?
</blockquote></details>

not for SHOW TYPES; i don't think it makes sense for creating composite types for example.


<!-- Sent from Reviewable.io -->

ajstorm reviewed (approved):
LGTM

otan commented:
bors r=ajstorm
